lib/data/external/GCP/GcpUtils.js

```diff
@@ -1 +1,110 @@
-module.exports = {};
+const werelogs = require('werelogs');
+const { errors } = require('arsenal');
+
+const _config = require('../../../Config').config;
+const { logHelper } = require('../utils');
+
+werelogs.configure({
+    level: _config.log.logLevel,
+    dump: _config.log.dumpLevel,
+});
+
+const logger = new werelogs.Logger('gcpUtil');
+
+class JsonError extends Error {
+    constructor(type, code, desc) {
+        super(type);
+        this.code = code;
+        this.description = desc;
+        this[type] = true;
+    }
+}
+
+function jsonRespCheck(err, resp, body, method, callback) {
+    if (err) {
+        logHelper(logger, 'error',
+            `${method}: error in json method`,
+            errors.InternalError.customizeDescription('json method failed'));
+        return callback(errors.InternalError
+            .customizeDescription('error in JSON Request'));
+    }
+    if (resp.statusCode >= 300) {
+        return callback(
+            new JsonError(resp.statusMessage, resp.statusCode));
+    }
+    let res;
+    try {
+        res = body && typeof body === 'string' ?
+            JSON.parse(body) : body;
+    } catch (error) { res = undefined; }
+    if (res && res.error && res.error.code >= 300) {
+        return callback(
+            new JsonError(res.error.message, res.error.code));
+    }
+    return callback(null, res);
+}
+
+function eachSlice(size) {
+    this.array = [];
+    let partNumber = 1;
+    for (let ind = 0; ind < this.length; ind += size) {
+        this.array.push({
+            Parts: this.slice(ind, ind + size),
+            PartNumber: partNumber++,
+        });
+    }
+    return this.array;
+}
+
+function getRandomInt(min, max) {
+    /* eslint-disable no-param-reassign */
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    /* eslint-enable no-param-reassign */
+    return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function createMpuKey(key, uploadId, partNumber, fileName) {
+    /* eslint-disable no-param-reassign */
+    if (typeof partNumber === 'string' && fileName === undefined) {
+        fileName = partNumber;
```

Comment:
For clarity, instead of reassigning params (which if sometimes useful is generally bad practice), I would recommend explicitly creating the ad-hoc key from the pristine params, like done at line 77.

Reply:
Will fix this.